virttest/utils_test/libvirt.py

```diff
@@ -3777,3 +3778,24 @@ def check_dumpxml(vm, content, err_ignore=False):
                 raise exceptions.TestFail("Expected '%s' was not found in "
                                           "%s's xml" % (content, vm.name))
     return True
+
+
+def checked_setup_or_cleanup_gluster(is_setup, vol_name, brick_path="", pool_name="",
+                                     file_path="/etc/glusterfs/glusterd.vol", **kwargs):
+    # pylint: disable=E1121
+    """
+    Set up or clean up glusterfs environment on localhost or remote
+    only if 'gluster_managed_by_test' is not "yes".
+
+    :param is_setup: Boolean value, true for setup, false for cleanup
+    :param vol_name: gluster created volume name
+    :param brick_path: Dir for create glusterfs
+    :param pool_name: if set, used to define brick path for pool
+    :param kwargs: Other parameters that need to set for gluster
+    :return: ip_addr of gluster instance (local or remote)
+    """
+    if not kwargs.get("gluster_managed_by_test", "yes") == "yes":
+        return kwargs.get("gluster_server_ip", utils_net.get_host_ip_address())
```

Comment:
The `utils_net.get_host_ip_address` is a bit complex function, so in this case it'd be better to call it only in case the `kwargs.get` fails:

```
gluster_server_ip = kwargs.get("gluster_server_ip")
if not gluster_server_ip:
    return utils_net.get_host_ip_address()
return gluster_server_ip
```

(the condition can as well be "is None" in case you want to allow "" returns)